### PR TITLE
PIR: Adding more test cases

### PR DESCRIPTION
**Pull request by shakyShane**
See https://app.asana.com/0/0/1206745203094838/f for more details.

This PR just alters _where_ we query for elements, so that unit-testing the logic is simpler/possible.

NOTE: no functional changes in this PR, just more tests.

### CODE REVIEW

**Comment by shakyShane** [comment]
* **#930** <a href="https://app.graphite.dev/github/pr/duckduckgo/content-scope-scripts/930?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @shakyShane and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: unit-test/broker-protection.js [hunk 1]
```diff
@@ -640,3 +646,29 @@ describe('utils', () => {
...
+            },
+            age: {
+                selector: '.name',
+                afterText: ','
+            }
+        }
+
+        const profile = createProfile(fac, data)
+        expect(profile.name).toEqual('John smith')
+        expect(profile.age).withContext('Age should be null here').toEqual(null)
```
Comment (shakyShane) [inline unit-test/broker-protection.js:672]: This is the failing test case!

File: unit-test/broker-protection.js [hunk 2]
```diff
@@ -640,3 +646,29 @@ describe('utils', () => {
...
+        const data = {
+            name: {
+                selector: '.name',
+                beforeText: ','
+            },
+            age: {
+                selector: '.name',
+                afterText: ','
+            }
+        }
```
Comment (shakyShane) [inline unit-test/broker-protection.js:668]: This format will now allow us to unit-test the extraction process from a high-level

File: src/features/broker-protection/actions/extract.js [hunk 3]
```diff
@@ -72,7 +77,12 @@ export function extractProfiles (action, userData, root = document) {
 
     return {
         results: profilesElementList.map((element) => {
-            const scrapedData = createProfile(element, action.profile)
+            const elementFactory = (key, value) => {
+                return value?.findElements
+                    ? cleanArray(getElements(element, value.selector))
+                    : cleanArray(getElement(element, value.selector) || getElementMatches(element, value.selector))
+            }
+            const scrapedData = createProfile(elementFactory, action.profile)
```
Comment (shakyShane) [inline src/features/broker-protection/actions/extract.js:85]: by moving the DOM-related operations into a function call like this, it means we can more easily test all the extraction logic without being tightly coupled to a real DOM.

File: unit-test/broker-protection-extract.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+                selector: '.name',
+                beforeText: ','
+            },
+            age: {
+                selector: '.name',
+                afterText: ','
+            }
+        }
+
+        const elementExamples = [
```
Comment (brianhall) [inline unit-test/broker-protection-extract.js:18]: So great to have test cases for this. It was such a guessing game when I first took the JSON files over!

**Review by brianhall — APPROVED** [review]
Looks great, really happy to have the ability to test the extraction logic more easily.

**Pull request merged**